app/controllers/discovered_hosts_controller.rb

```diff
@@ -289,6 +290,21 @@ def find_by_name(*includes)
     @host
   end
 
+  def check_for_subnet
+    hosts_without_subnet = []
+    case params[:action]
+      when 'reboot', 'auto_provision'
+        if @host.subnet.nil?
+          process_warning :warning_msg => _("Discovered host reported from unknown subnet, communication will not be proxied."), :warning_redirect => request.referer
+        end
+      when 'submit_multiple_reboot', 'submit_multiple_auto_provision'
+        if @hosts.any? { |host| host.subnet.nil? }
```

Comment:
I think you can use `select` line directly instead of using .any? then `.select`.
With `host_names_without_subnet =  @hosts.select { |host| host.subnet.nil?}.pluck(:name)` you will get list of names which you could use this for checking condition.